### PR TITLE
Custom Reports AutoComplete

### PR DESCRIPTION
**Pull request by carkom**
Creating an autocomplete for custom reports so they can be recalled without switching back to the dashboard.

### CODE REVIEW

**Comment by Teprifer** [comment]
I have this feeling you probably have some of these on your list already... Apologies if it comes across nit picky!

* Input box isn't grabbing focus when it's displayed (saving report gets focus).

* Escape key isn't dismissing the pop-up (does for the others in this menu).

* Suggestion, change the word 'Apply' to indicate the action a bit better, e.g. 'load', "Load Report', 'Open', 'Open report' (you get the idea).

* I had this suggestion, then I kinda thought how easy it is it setup a report so I'm unsure, as it could also get annoying if you liked to toggle stuff on one report, and if added here, it would somewhat have to be added when using the browser or actual budget back buttons: Warning prompt if leaving a report with unsaved changes. But not on 'unsaved report' to stop quick checks being annoying?

* Suggestion: smooth out switching reports by loading the report on selection, instead of having to select it, then confirm the selection.

* Crash if no input is provided.

`toe@https://deploy-preview-2350.demo.actualbudget.org/static/js/ReportRouter.9ZZ2mKcf.chunk.js:113:70855 T1@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:56:19539 W0@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:3139 gD@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:44820 fD@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:39779 XL@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:39707 Qh@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:39557 tv@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:35921 Ek@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:36725 ma@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:56:3279 ir/<@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:34250 `

* Crash if a value not matching a report name is entered (same thing as nothing eh).

`toe@https://deploy-preview-2350.demo.actualbudget.org/static/js/ReportRouter.9ZZ2mKcf.chunk.js:113:70855 T1@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:56:19539 W0@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:3139 gD@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:44820 fD@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:39779 XL@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:39707 Qh@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:39557 tv@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:35921 Ek@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:36725 ma@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:56:3279 ir/<@https://deploy-preview-2350.demo.actualbudget.org/static/js/index.Icr74xxU.js:58:34250`

File: packages/desktop-client/src/components/reports/SaveReport.tsx [hunk 1]
```diff
@@ -145,15 +160,14 @@ export function SaveReport({
             flexShrink: 0,
           }}
         >
-          {!report.id ? 'Unsaved report' : report.name}&nbsp;
+          {report.id === '' ? 'Unsaved report' : report.name}&nbsp;
```
Comment (twk3) [inline packages/desktop-client/src/components/reports/SaveReport.tsx:163]: Did we need this to change? The previous covered all falsey including empty string.

File: packages/desktop-client/src/components/reports/reports/CustomReport.jsx [hunk 2]
```diff
@@ -308,6 +308,27 @@ export function CustomReport() {
...
+        setBalanceType(savedReport.balanceType);
+        setShowEmpty(savedReport.showEmpty);
+        setShowOffBudget(savedReport.showOffBudget);
+        setShowHiddenCategories(savedReport.showHiddenCategories);
+        setShowUncategorized(savedReport.showUncategorized);
+        setSelectedCategories(savedReport.selectedCategories);
+        setGraphType(savedReport.graphType);
+        onApplyFilter(null);
+        savedReport.conditions.forEach(condition => onApplyFilter(condition));
+        onCondOpChange(savedReport.conditionsOp);
```
Comment (twk3) [inline packages/desktop-client/src/components/reports/reports/CustomReport.jsx:330]: suggestion: move these sets to a function that we can use here, and for reload, as well as the reset.

File: packages/desktop-client/src/components/reports/SaveReport.tsx [hunk 3]
```diff
@@ -145,15 +160,14 @@ export function SaveReport({
...
         </Text>
         {savedStatus === 'modified' && <Text>(modified)&nbsp;</Text>}
         <SvgExpandArrow width={8} height={8} style={{ marginRight: 5 }} />
       </Button>
       {menuOpen && (
         <SaveReportMenu
```
Comment (twk3) [inline packages/desktop-client/src/components/reports/SaveReport.tsx:169]: idea: would be nice to disable the choose reports button if we don't have any saved reports.

File: packages/desktop-client/src/components/reports/reports/CustomReport.jsx (hunk 2 above)
Comment (carkom) [reply packages/desktop-client/src/components/reports/reports/CustomReport.jsx:336]: Yea, good call.

**Review by twk3 — APPROVED** [review]
Changes look good

and I didn't run into anything unexpected related to the change while using it in the interface

:+1:

**Pull request merged**